var/spack/repos/builtin/packages/exago/package.py

```diff
@@ -62,6 +62,55 @@ class Exago(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("cmake@3.18:", type="build")
 
+    # Profiling
+    depends_on(
+        "hiop+deepchecking build_type=RelWithDebInfo", when="+hiop build_type=RelWithDebInfo"
+    )
+    depends_on("hiop~deepchecking  build_type=Release ", when="+hiop build_type=Release ")
+
+    # Control the package's build-type depending on the release or debug flag
+    for pkg in ["raja", "umpire", "magma", "camp"]:
+        depends_on("{0} build_type=Release".format(pkg), when="build_type=Release")
+        depends_on("{0} build_type=RelWithDebInfo".format(pkg), when="build_type=RelWithDebInfo")
+
+    # Force RelWithDebInfo when not using release
+    conflicts(
+        "build_type=Release",
+        when="build_type=RelWithDebInfo",
+        msg="Use RelWithDebInfo when not using release build type",
+    )
+    conflicts(
+        "build_type=Debug",
+        when="build_type=RelWithDebInfo",
+        msg="Use RelWithDebInfo when not using release build type",
+    )
+    conflicts(
+        "build_type=MinSizeRel",
+        when="build_type=RelWithDebInfo",
+        msg="Use RelWithDebInfo when not using release build type",
+    )
+
+    # Force Release mode when using optimizations
+    conflicts(
+        "build_type=RelWithDebInfo",
+        when="build_type=Release",
+        msg="Use Release when using release build type",
+    )
+    conflicts(
+        "build_type=Debug",
+        when="build_type=Release",
+        msg="Use Release when using release build type",
+    )
+    conflicts(
+        "build_type=MinSizeRel",
+        when="build_type=Release",
+        msg="Use Release when using release build type",
+    )
```

Comment:
These conflicts mean nothing anymore, and should be removed. 

I guess we were significantly over complicating things initially, but I like the solution we've arrived at.